### ISSUE TITLE
Bad advice in an unsafe world

### ISSUE THREAD

**Issue by greg-ski** [MicrosoftDocs/OfficeDocs-Exchange#367]
The advice that "If you have firewalls or network devices that could potentially restrict or alter this kind of network traffic, you need to configure rules that allow free and unrestricted communication between these servers (rules that allow incoming and outgoing network traffic on any port—including random RPC ports—and any protocol that never alter bits on the wire)." is very bad advice in this day and age. This endorses behaviours within app teams that do not know any better and creates problems for organisations that are attempting to be security conscious. This advice needs to be revised and the original content needs to either be taken down or publicly refuted.

I look forward to a more appropriate edition to this content anon. 

Yours
G Malewski
Infrastructure and Security Architect

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2c06c27d-22fa-911b-db81-233ee4a16529
* Version Independent ID: c9136fcc-9cd7-d36a-1097-f3b82265f68a
* Content: [Network ports for clients and mail flow in Exchange](https://docs.microsoft.com/en-us/Exchange/plan-and-deploy/deployment-ref/network-ports?view=exchserver-2019)
* Content Source: [Exchange/ExchangeServer/plan-and-deploy/deployment-ref/network-ports.md](https://github.com/MicrosoftDocs/OfficeDocs-Exchange/blob/public/Exchange/ExchangeServer/plan-and-deploy/deployment-ref/network-ports.md)
* Product: **exchange-server-it-pro**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
I added the "internal network traffic" clarification as described.

**Comment by greg-ski** [comment]
Hi Chris,
Apologies for the delay in responding. I understand your motivations and
foresaw your response. That said, a broad 'any any' rule sends the wrong
message: it is an optics issue especially with regards to connectivity with
domain controllers for which there is a well established pattern. The
criticality of these services is very high given how central they are
within an organisations IT landscape.

My thinking is that a better approach would have been for the article to:
 - refer to and follow the generally understood pattern for connectivity
with domain controllers
 - call out the set of services that are used between internal servers
 - suggest a minimum viable set of flows. An example of which might be:
src=exchange_servers, dst=exchange_servers,
dstport=portx,porty portz,49152-65535
The latter of which cover the services on fixed ports and those that appear
on ephemeral ports e.g. RPC
 - provide recommendation of which flows can be tightened more, and what
constraints thete may be e.g. the firewall must be able to understand the
port mapper protocol.

I haven't been able to readily find anything online that lists each
service, their ports, or descriptions so I've had to leave the above
example as a generic one but I'd be willing to  collaborate to see what is
possible.

Your thoughts on the above would be most appreciated

Regards,
Greg

**Comment by chrisda** [comment]
@greg-ski, I respect what you're saying, but I was given a very clear mandate for that language when I wrote the topic for 2013. Apparently, Microsoft Support had massive problems in the past with customers attempting to restrict internal traffic based on the honesty in earlier Exchange port reference topics that document the ports used between internal Exchange servers and DCs. We said "these are the ports used" so customers thought it was OK to block everything else -- even though we said not to.

The statement that is apparently missing is "Exchange was designed to have unrestricted access between internal Exchange servers and internal Exchange servers and internal Domain Controllers." I thought that phrase was in there; that was definitely the sentiment I got when I was creating the topic. Does adding that help?
